app/models/jobs-model.js

```diff
@@ -9,16 +10,25 @@ const sortRef = {
   employer: { field: 'employer', direction: 'ASC' },
 };
 
+const historicDate = {
+  week: moment().subtract(moment.duration(1, 'w')),
+  fortnight: moment().subtract(moment.duration(2, 'w')),
+  month: moment().subtract(moment.duration(1, 'M')),
+  none: moment('2016-01-01'),
+};
+
 module.exports = db.Model.extend(
   {
     tableName: 'jobs',
     hasTimestamps: true,
   },
   {
-    findAllByAccountId(accountId, sort) {
+    findAllByAccountId(accountId, sort, filter) {
       const sortOrDefault = (sort == null || sort === '') ? 'created' : sort;
+      const filterOrDefault = (filter == null || filter === '') ? 'none' : filter;
```

Comment:
it is because:
you don't want this `const sortOrDefault =  sort  ? 'created' : sort`
you want this `const sortOrDefault =  sort  ? sort: 'created' ` or even better `const sortOrDefault = sort || 'created'`